tasks/sign-binaries/sign-binaries.yaml

```diff
@@ -42,36 +45,203 @@ spec:
+              key: app_specific_password
+        - name: QUAY_USER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: username
+        - name: QUAY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: password
+      script: |
+        #!/usr/bin/env bash
+        set -eux
 
-    # - name: sign-mac-binaries
-    #   image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
-    #   script: |
-    #     #!/usr/bin/env bash
-    #     # TODO CLDX-79
+        mkdir -p /root/.ssh
+        chmod 700 /root/.ssh
+        cp "/etc/secrets/mac_id_rsa" /root/.ssh/id_rsa
+        cp "/etc/secrets/mac_fingerprint" /root/.ssh/known_hosts
+        chmod 600 root/.ssh/known_hosts /root/.ssh/id_rsa
+
+        #TODO: cp SSH key to the container
+        # shell check complains about the variable (unsigned_digest) not being used but it is used in the script
+        # shellcheck disable=SC2034
+        unsigned_digest=$(cat "$(results.unsignedMacDigest.path)")
+        mac_signing_script="/tmp/mac_signing_script.sh"
+        set +x
+        cat << 'EOF' > "$mac_signing_script"
+        #!/bin/bash
+        set -e 
+        TEMP_DIR="/tmp/$(params.pipelineRunUid)"
+        BINARY_PATH="binaries_dir"
+        ZIP_PATH="$TEMP_DIR/signed_content.zip"
+        DIGEST_FILE="$TEMP_DIR/push_digest.txt"
+
+        cd "$TEMP_DIR"
+        oras login quay.io -u ${QUAY_USER} -p ${QUAY_PASS}
+        oras pull $(params.quayURL)@sha256:$unsigned_digest -o "$TEMP_DIR/$BINARY_PATH"
+        # This is the directory where the content was extracted
+        CONTENT_DIR=$(find "$BINARY_PATH" -maxdepth 1 -type d | tail -n 1)
+
+        security unlock-keychain -p $KEYCHAIN_PASSWORD login.keychain
+
+        echo "Signing files in the $CONTENT_DIR directory..."
+        find "$CONTENT_DIR" -type f | while read file; do
+            echo "Signing: $file"
+            if ! xcrun codesign --sign "Developer ID Application: $SIGNING_IDENTITY" \
+                --options runtime --timestamp --force "$file"; then
+                echo "Failed to sign file: $file"
+                exit 1
+            fi
+        done
+
+        cd "$BINARY_PATH"
+        CONTENT_DIR=$(basename "$CONTENT_DIR")
+        zip -r "$ZIP_PATH" "$CONTENT_DIR"
+
+        xcrun notarytool submit "$ZIP_PATH" \
+            --wait \
+            --apple-id "$APPLE_ID" \
+            --team-id "$TEAM_ID" \
+            --password "$APP_SPECIFIC_PASSWORD"
+
+        PUSH_OUTPUT=$(oras push "$(params.quayURL)/signed:$(params.pipelineRunUid)" "$CONTENT_DIR")
+        SIGNED_DIGEST=$(echo "$PUSH_OUTPUT" | grep 'Digest:' | awk '{print $2}')
+        echo "$SIGNED_DIGEST" >> "$DIGEST_FILE"
+        echo "Process completed successfully."
+        EOF
+        set -x
+        # Copy the script to the Mac host
+        scp -i "/etc/secrets/id_rsa" "$mac_signing_script" \
```

Comment:
Why do you need the `-i`? You copied the private key to `/root/.ssh/id_rsa` so that should work, right? Also, it will probably not work from this location since it won't have the required permissions. This applies to the others as well.

Reply:
When I was running tests for the Windows tasks, `whoami` from the container runs returned `root` but the home directory for root was `/home/tekton`

During testing when I put the id_rsa into `~/.ssh` or `/home/tekton/.ssh` it wouldn't find it by default. (and it should have per the ssh_config)

But when I put it at `/root/.ssh/` it worked fine.

Maybe the containers run with a different user in prod? My tests were all just ran locally and when I exec into a running container and look at /etc/passwd I didn't see any other users..

Reply:
OK, yeah, I guess whatever works is fine. But it doesn't make sense to copy the key to another place, but then not use that.

Reply:
Yep I agree. This should be `scp -i "/root/.ssh/id_rsa" `